calculator.js

```diff
@@ -1,2 +1,32 @@
 var calculator = {
-}
+  add: function(x,y) {
+    console.log(x+y);
+    return x+y;
+  },
+  subtract: function(x,y) {
+    console.log(x-y);
+    return x-y;
+  },
+  multiply: function(x,y) {
+    console.log(x*y);
+    return x*y;
+  },
+  divide: function(x,y) {
+    console.log(x/y);
+    return x/y;
+  }
+  exponent: Math.pow(x,y) {
+    console.log(x^,y^);
+    return x^y^;
+  },
+  remainder: function(x,y) {
+    console.log(x%y);
+    return(x%y);
+  }
+// var remValue = (x,y)------> bonus
```

Comment:
I'd concentrate on testing your code rather than getting the bonuses, but the first bonus just needs a variable to store the most recent calculation. For the 2nd bonus, I'll give you a hint: you'll need to use a certain javascript function that *evaluates* code...